ls1intum/Artemis
# `General`: Allow admins to hide the password-based login form

### Checklist
#### General


#### Server
#### Client

### Motivation and Context



@LDAP and @dfuchss  at KIT aim to completely disable password login and rely only on SAML2. Having a password login causes confusion since Artemis credentials do not match SAML2 credentials.
When both authentication options were enabled, it is not obvious to users that when using Single Sign-on they do not have to put anything into the login form.

### Description


A new configuration option `info.saml2.password-login-disabled` was added to the SAML2 config file.
I implemented the client changes, so that this configuration option disables the traditional login form. I also designed the user interface to now make it clear that there are two options to sign in. The UI is responsive and adjusts itself based on the admin's configuration.
The query parameter `?showLoginForm` overrides the configuration and displays the password-based login nevertheless (useful for admin login).
Furthermore, I added the `info.saml2.idenitity-provider-name` configuration setting to optionally add the name of the identity provider to the title on the sign-in page.

### Steps for Testing

Prerequisites:
- 1 Administrator

1. Start Artemis locally with SAML2 authentication enabled
2. Make sure both sign on options are visible, working and UI is responsive
3. Set `info.saml2.password-login-disabled: true` in your local configuration
4. Ensure only SSO authentication is visible
5. Adding the query parameter`?showLoginForm` to the URL should also display the traditional login form (override)
6. Disable SAML2 authentication, the password form should appear even though `password-login-disabled` is set (fallback)

### Review Progress





#### Code Review
#### Manual Tests

### Screenshots



Both traditional and SSO authentication enabled:

![Screenshot 2022-06-26 at 16-35-24 Welcome to Artemis!](https://user-images.githubusercontent.com/102237363/175820036-3837354d-af77-4d21-998b-2ddf0afcb46b.png)

Only password-based sign on:

![Screenshot 2022-06-26 at 16-17-40 Welcome to Artemis!](https://user-images.githubusercontent.com/102237363/175820052-963f61b4-aaa6-4524-9032-e0f1f47b611d.png)

Only SSO (e.g., SAML2) authentication:

![Screenshot 2022-06-26 at 16-16-51 Welcome to Artemis!](https://user-images.githubusercontent.com/102237363/175820046-fe317bec-798e-4678-9172-c5ad9cd062ca.png)
